src/bindings/python/flux/job/executor.py

```diff
@@ -193,16 +193,22 @@ def jobid(self, timeout=None):
         :param timeout: The number of seconds to wait for the jobid.
             If None, then there is no limit on the wait time.
 
-        :return: a positive integer jobid, or ``-1`` if the future was cancelled.
+        :return: a positive integer jobid.
 
         :raises concurrent.futures.TimeoutError: If the jobid is not available
             before the given timeout.
+        :raises concurrent.futures.CancelledError: If the future was cancelled.
         """
         if self.__jobid is not None:
+            # if the future is cancelled the jobid is set to an exception
+            if isinstance(self.__jobid, concurrent.futures.CancelledError):
+                raise self.__jobid
```

Comment:
Would it be clearer to just `raise CancelledError` here? The current approach seems fine as well (though it is confusing the LGTM check as well). I guess I'm curious if there's extra context in the `self.__jobid` object that prevents you from raising a `CancelledError` directly. (reminder: I'm not all that familiar with Python and its idioms)

Reply:
Yeah I think that would probably be better. There's no extra context that I know of. I could also change the ``if isinstance()`` check to ``if self.cancelled()`` which is probably clearer.